src/sentry/native/native_sdk.cpp

```diff
@@ -0,0 +1,207 @@
+	sentry_value_t event = sentry_value_new_event();
+	sentry_value_set_by_key(event, "level",
+			sentry_value_new_string(sentry::level_as_cstring(p_level)));
+
+	sentry_value_t exception = sentry_value_new_exception(p_type.utf8(), p_value.utf8());
+	sentry_value_t stack_trace = sentry_value_new_object();
+
+	sentry_value_t frames = sentry_value_new_list();
+	for (const StackFrame &frame : p_frames) {
+		sentry_value_t sentry_frame = sentry_value_new_object();
+		sentry_value_set_by_key(sentry_frame, "filename", sentry_value_new_string(frame.filename.utf8()));
+		sentry_value_set_by_key(sentry_frame, "function", sentry_value_new_string(frame.function.utf8()));
+		sentry_value_set_by_key(sentry_frame, "lineno", sentry_value_new_int32(frame.lineno));
+		if (!frame.context_line.is_empty()) {
+			sentry_value_set_by_key(sentry_frame, "context_line", sentry_value_new_string(frame.context_line.utf8()));
+			sentry_value_set_by_key(sentry_frame, "pre_context", sentry::native::strings_to_sentry_list(frame.pre_context));
+			sentry_value_set_by_key(sentry_frame, "post_context", sentry::native::strings_to_sentry_list(frame.post_context));
+		}
+		sentry_value_append(frames, sentry_frame);
+	}
+
+	sentry_value_set_by_key(stack_trace, "frames", frames);
+	sentry_value_set_by_key(exception, "stacktrace", stack_trace);
+	sentry_event_add_exception(event, exception);
+	sentry_capture_event(event);
+}
+
+void NativeSDK::initialize() {
+	ERR_FAIL_NULL(OS::get_singleton());
+	ERR_FAIL_NULL(ProjectSettings::get_singleton());
+
+	sentry_options_t *options = sentry_options_new();
+	sentry_options_set_dsn(options, SentryOptions::get_singleton()->get_dsn());
+
+	// Establish handler path.
+	String handler_fn;
+#ifdef LINUX_ENABLED
+	handler_fn = "crashpad_handler";
+#elif WINDOWS_ENABLED
+	handler_fn = "crashpad_handler.exe";
+#elif MACOS_ENABLED
+	handler_fn = "crashpad_handler";
+#endif
+	String handler_path = OS::get_singleton()->get_executable_path().get_base_dir() + "/" + handler_fn;
+	if (!FileAccess::file_exists(handler_path)) {
+		handler_path = ProjectSettings::get_singleton()->globalize_path("res://addons/sentrysdk/bin/" + handler_fn);
+	}
+	if (FileAccess::file_exists(handler_path)) {
+		sentry_options_set_handler_path(options, handler_path.utf8());
+	} else {
+		ERR_PRINT(vformat("Sentry: Failed to locate crash handler (crashpad) - backend disabled (%s)", handler_path));
+		sentry_options_set_backend(options, NULL);
+	}
+
+	sentry_options_set_database_path(options, (OS::get_singleton()->get_user_data_dir() + "/sentry").utf8());
+	sentry_options_set_sample_rate(options, SentryOptions::get_singleton()->get_sample_rate());
+	sentry_options_set_release(options, SentryOptions::get_singleton()->get_release());
+	sentry_options_set_debug(options, SentryOptions::get_singleton()->is_debug_enabled());
+	sentry_options_set_environment(options, sentry::environment::get_environment());
+	sentry_options_set_sdk_name(options, "sentry.native.godot");
```

Comment:
We usually have a subset of events come through sentry.native so having it report the original version (of sentry-native) made sense.

In this, since all events of this SDK will be `sentry.native.godot` I suspect it's better if we override the version and have this SDK's version. So when we get a bug report, it's easy to know what folks are using. Otherwise we might make changes on this SDK withotu bumping sentry-native and it won't be possible for us to tell them apart.

Is `sentry_options_set_sdk_version` available? Otherwise we'll need to add it.

This would also apply to Unreal which wraps native on desktop, @tustanivsky how does it work there?

Reply:
Knowing sentry-native version seems useful too. Is it viable to have both versions reported?

Reply:
We'll know the sentry-native version by knowing which Godot SDK version it is. Since the versions here are tagged here and dep version is fixed

Reply:
Granted it's not as easy as just looking at the event in Sentry to figure out, but today we only have 1 version